### PR TITLE
[wip] testing kube-scheduler api build

### PR DESCRIPTION
**Pull request by kbhawkey**
Initial testing of building kube-scheduler API ref

### CODE REVIEW

**Comment by kbhawkey** [comment]
See issue #138

**Comment by kbhawkey** [comment]
This is not automated yet. Tested locally.
/cc @tengqm

**Comment by kbhawkey** [comment]
See PR https://github.com/kubernetes/website/pull/19903

File: gen-kube-scheduler/kube-scheduler-api-ref.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Change this to installation directory of gen-crd-api-reference-docs
+GEN_DOCS=/tmp/gen-crd-api-reference-docs
+
+# Change this to where the website repository is cloned.
+K8S_WEBSITE=/website
+
+K8S_SCHEDULER=k8s.io/kube-scheduler
+
+# table style substitutions
+TABLE_SUB='<div class=\"table-responsive\"><table class=\"table table-bordered\">'
```
Comment (tengqm) [inline gen-kube-scheduler/kube-scheduler-api-ref.sh]: would this bootstrap classes work in the generated file? I don't see any reference to the bootstrap CSS.

File: gen-kube-scheduler/kube-scheduler-api-ref.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+THEAD_SUB='<thead class=\"thead-light\">'
+
+# FIX, generate in reference docs
+CONTENT_DIR=kubernetes-sigs/reference-docs
+
+echo "+++
```
Comment (tengqm) [inline gen-kube-scheduler/kube-scheduler-api-ref.sh]: +++ or --- ?

File: gen-kube-scheduler/kube-scheduler-api-ref.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+title = \"Kube Scheduler API Reference\"
+description = \"Reference documentation for Kube Scheduler\"
```
Comment (tengqm) [inline gen-kube-scheduler/kube-scheduler-api-ref.sh]: quotes on line 26-27 are unnecessary?

File: gen-kube-scheduler/kube-scheduler-api-ref.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# git clone https://github.com/kubernetes/kubernetes.git
+
+# copied staging, src/k8s.io/kube-scheduler
+# /tmp/tmpdc3iihix/src/k8s.io/kubernetes/staging/src/k8s.io/kube-scheduler
+
+# Change this to installation directory of gen-crd-api-reference-docs
+GEN_DOCS=/tmp/gen-crd-api-reference-docs
+
+# Change this to where the website repository is cloned.
+K8S_WEBSITE=/website
```
Comment (tengqm) [inline gen-kube-scheduler/kube-scheduler-api-ref.sh]: This is currently not used?

File: gen-kube-scheduler/kube-scheduler-api-ref.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+weight = 100
++++" > ${CONTENT_DIR}/common.md
+
+${GEN_DOCS}/gen-crd-api-reference-docs -api-dir ./config -config ${GEN_DOCS}/example-config.json -template-dir ${GEN_DOCS}/template -out-file temp_common.md
+cat temp_common.md >> ${CONTENT_DIR}/common.md
+rm temp_common.md
+
+sed 's/<table>/'"$TABLE_SUB"'/g' ${CONTENT_DIR}/common.md > temp_common.md
+sed 's/<thead>/'"$THEAD_SUB"'/g' temp_common.md > ${CONTENT_DIR}/common.md
+rm temp_common.md
```
Comment (tengqm) [inline gen-kube-scheduler/kube-scheduler-api-ref.sh]: we could instead perform the substitution (35-36) on the temp_common.md file generated from line 31, before deleting it

File: gen-kube-scheduler/kube-scheduler-api-ref.sh (hunk 2 above)
Comment (kbhawkey) [reply gen-kube-scheduler/kube-scheduler-api-ref.sh]: Yes

File: gen-kube-scheduler/kube-scheduler-api-ref.sh (hunk 1 above)
Comment (kbhawkey) [reply gen-kube-scheduler/kube-scheduler-api-ref.sh]: Yes, possibly. I plan to close this PR for now and write up the steps I took in the issue.

**Pull request closed without merging**